File: structural_pattern/composite/File.java
package composite;

public class File implements FileComponent {
    private String name;
    private int size;

    public File(String name, int size) {
        this.name = name;
        this.size = size;
    }

    @Override
    public void showDetails(int indent) {
        System.out.println(" ".repeat(indent) + "📄File: " + name + " (" + size + " KB");
    }

    @Override
    public int getSize() {
        return size;
    }

    @Override
    public String getName() {
        return name;
    }
}

File: typescript/structural_pattern/src/composite/File.ts
import FileComponent from "./FileComponent";

class File implements FileComponent {
    private name: string;
    private size: number;

    public constructor(name: string, size: number) {
        this.name = name;
        this.size = size;
    }

    public showDetails(indent: number): void {
        console.log("   ".repeat(indent) + "📄File: " + this.name + " (" + this.size + " KB)");
    }

    public getSize(): number {
        return this.size;
    }

    public getName(): string {
        return this.name;
    }
}

export default File;